src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -361,6 +504,69 @@ double KevinHallModel::get_weight_quantile(double epa_quantile, core::Gender sex
     return weight_quantiles_.at(sex)[weight_index];
 }
 
+KevinHallAdjustmentTable KevinHallModel::compute_mean_weight(Population &population,
+                                                             std::optional<double> power) const {
+
+    // Local struct to hold count and sum of weight powers.
+    struct SumCount {
+      public:
+        void append(double value) noexcept {
+            sum_ += value;
+            count_++;
+        }
+
+        double mean() const noexcept { return sum_ / count_; }
+
+      private:
+        double sum_{};
+        int count_{};
+    };
+
+    // Compute sums and counts of weight powers for sex and age.
+    auto sumcounts = UnorderedMap2d<core::Gender, int, SumCount>{};
+    sumcounts.emplace_row(core::Gender::female, std::unordered_map<int, SumCount>{});
+    sumcounts.emplace_row(core::Gender::male, std::unordered_map<int, SumCount>{});
+    for (const auto &person : population) {
+        if (!person.is_active()) {
+            continue;
+        }
+
+        double value;
+        if (power.has_value()) {
+            value = pow(person.risk_factors.at("Weight"_id), power.value());
```

Comment:
Out of interest, can powers be floating-point numbers or are they always integers?

Reply:
Strictly they're the the type of the base and return. Either `float` or `double`.

Reply:
I meant for *this* code not in general. Are you using fractional powers or just integers? Because if it's just integers, this bit could be replaced with a `for`-loop.

Reply:
For part of the model, the height initialise and update I think, a real-valued `slope` parameter is used.